src/Shared/Microsoft.DotNet.Kusto/KustoClientProvider.cs

```diff
@@ -87,6 +81,106 @@ public async Task<IDataReader> ExecuteKustoQueryAsync(KustoQuery query)
             }
         }
 
+        /// <summary>
+        /// Use this method to receive large quantities of data from Kusto in a "stream". 
+        /// This method assumes the query is returning a single schema result set. 
+        /// </summary>
+        /// <param name="query"></param>
+        /// <returns></returns>
+        public IEnumerable<object[]> ExecuteStreamableKustoQuery(KustoQuery query)
+        {
+            var client = GetProvider();
+            var properties = BuildClientRequestProperties(query);
+            properties.SetOption(ClientRequestProperties.OptionResultsProgressiveEnabled, true);
+
+            string text = BuildQueryText(query);
+
+            ProgressiveDataSet pDataSet = client.ExecuteQueryV2Async(
+                DatabaseName,
+                text,
+                properties).GetAwaiter().GetResult();
+
+            int tableCompletionCount = 0;
+
+            using (var frames = pDataSet.GetFrames())
```

Comment:
You could use a `using statement` here, and not need to indent everything. Something like 
```suggestion
            using var frames = pDataSet.GetFrames();
```
And expand whatever the "var" is into the full type (since that's what our style guidelines say to do).  I think it's an "IEnumerator<SOMETHING>"?